Add 6.2.0 release notes (#2903)
sensu/sensu-docs: content/sensu-go/6.2/release-notes.md

---

## 6.2.0 release notes

**December 17, 2020** &mdash; The latest release of Sensu Go, version 6.2.0, is now available for download.

> **ADD RELEASE SYNOPSIS HERE**

Author: ```suggestion
The latest release of Sensu Go, version 6.2.0, is now available for download! Sensu Go 5.x and configuration management users rejoice: this release adds support for agent local configuration (i.e., agent.yaml) managed entities! Agent entities may now be managed exclusively by their agents when `sensu-agent` is started with the new `agent-managed-entity` configuration option. This makes it more straightforward to migrate from Sensu Go 5.x to 6.x, as existing agent entity management workflows (i.e., Puppet) will just work with the new option enabled! Note that you will not be able to edit agent-managed entities via the backend REST API or web UI. This release also includes significant feature enhancements including PostgreSQL backend round robin check scheduling for better reliability and consistency. Also, the prune API no longer requires cluster-wide permissions; users with limited permissions can put it to use in their namespaces!
```

See the [upgrade guide][1] to upgrade Sensu to version 6.2.0.

**NEW FEATURES:**

- ([Commercial feature][193]) Added support for the `memberof` attribute for the [LDAP authentication provider][199].
- ([Commercial feature][193]) Added the ability to exclude resource types when using sensuctl prune with the [--omit flag][200].
- ([Commercial feature][193]) Added support for [round robin scheduling on PostgreSQL][201] instead of etcd.
- ([Commercial feature][193]) Added support for OIDC authentication via [sensuctl configure][202].
- The [/metrics API endpoint][196] now exposes build information as a Prometheus metric.

> - Added /health endpoint to agentd.

Reviewer: ```suggestion
- Added /health API endpoint to agent websocket transport.
```

Author: Good catch...don't know why I thought anyone would put together what the heck I was talking about there :D

> The latest release of Sensu Go, version 6.2.0, is now available for download! Sensu Go 5.x and configuration management users rejoice: this release adds support for agent local configuration (i.e. agent.yaml) managed entities! Agent entities may now be managed exclusively by their agents when `sensu-agent` is started with the new `agent-managed-entity` configuration option. This makes it more straightforward to migrate from Sensu Go 5.x to 6.x, as existing agent entity management workflows like Puppet will just work with the new option enabled! Note that you will not be able to edit agent-managed entities via the backend REST API or web UI. This release also includes significant feature enhancements such as PostgreSQL backend round robin check scheduling for better reliability and consistency. Also, the prune API no longer requires cluster-wide permissions; users with limited permissions can put it to use in their namespaces!

Reviewer: ```suggestion
The latest release of Sensu Go, version 6.2.0, is now available for download! Sensu Go 5.x and configuration management users rejoice: this release adds support for agent local configuration (i.e. agent.yml) managed entities! Agent entities may now be managed exclusively by their agents when `sensu-agent` is started with the new `agent-managed-entity` configuration option. This makes it more straightforward to migrate from Sensu Go 5.x to 6.x, as existing agent entity management workflows like Puppet will just work with the new option enabled! Note that you will not be able to edit agent-managed entities via the backend REST API or web UI. This release also includes significant feature enhancements such as PostgreSQL backend round robin check scheduling for increased reliability and consistency. Also, the prune API no longer requires cluster-wide permissions; users with limited permissions can put it to use in their namespaces!
```

agent.yaml -> agent.yml

better -> increased

> The latest release of Sensu Go, version 6.2.0, is now available for download! Sensu Go 5.x and configuration management users rejoice: this release adds support for agent local configuration (i.e., agent.yaml) managed entities! Agent entities may now be managed exclusively by their agents when `sensu-agent` is started with the new `agent-managed-entity` configuration option. This makes it more straightforward to migrate from Sensu Go 5.x to 6.x, as existing agent entity management workflows (i.e., Puppet) will just work with the new option enabled! Note that you will not be able to edit agent-managed entities via the backend REST API or web UI. This release also includes significant feature enhancements including PostgreSQL backend round robin check scheduling for better reliability and consistency. Also, the prune API no longer requires cluster-wide permissions; users with limited permissions can put it to use in their namespaces!

Author: ```suggestion
The latest release of Sensu Go, version 6.2.0, is now available for download! Sensu Go 5.x and configuration management users rejoice: this release adds support for agent local configuration (i.e. agent.yaml) managed entities! Agent entities may now be managed exclusively by their agents when `sensu-agent` is started with the new `agent-managed-entity` configuration option. This makes it more straightforward to migrate from Sensu Go 5.x to 6.x, as existing agent entity management workflows like Puppet will just work with the new option enabled! Note that you will not be able to edit agent-managed entities via the backend REST API or web UI. This release also includes significant feature enhancements such as PostgreSQL backend round robin check scheduling for better reliability and consistency. Also, the prune API no longer requires cluster-wide permissions; users with limited permissions can put it to use in their namespaces!
```

- Checks now include the [`scheduler` attribute][197], which Sensu automatically populates with the type of scheduler that schedules the check.
- Events now include the [`sequence` attribute][198], which the Sensu agent automatically sets at startup and increments by 1 at every successive check execution or keepalive event.
- Added support for using environment variables to define the configuration file paths for the Sensu agent (`SENSU_CONFIG_FILE`) and backend (`SENSU_BACKEND_CONFIG_FILE`).

**IMPROVEMENTS:**

- ([Commercial feature][193]) Refactored entity limiter to ensure that warning messages about approaching a license's entity or entity class limit are now only displayed for users with `create` or `update` permissions for the license.

> - Adjusted the format for silenced entry dates and durations in sensuctl. For all silenced entries, the begin date is now listed in RFC 3339 format. For silenced entries that have not begun, the list displays the expiration date in RFC 3339 format. For silenced entires with no expiration date, the list displays `-1`. For silenced entries that have begun, the list displays the duration (e.g. 1m30s).

Reviewer: ```suggestion
- Adjusted the format for silenced entry dates and durations in sensuctl tabular-format output. For all silenced entries, the begin date is now listed in RFC 3339 format. For silenced entries that have not begun, the list displays the expiration date in RFC 3339 format. For silenced entires with no expiration date, the list displays `-1`. For silenced entries that have begun, the list displays the duration (e.g. 1m30s).
```

Reviewer: Ideally we would link to an explanation of RFC 3339, but the most obvious route of linking to Wikipedia winds up at https://en.wikipedia.org/wiki/ISO_8601 which might be even more confusing. 🤔

Author: Yeah. There is also the [golang time page](https://golang.org/pkg/time/#pkg-constants), but there's no good heading for linking straight to RFC3339 (readers would have to dig)

- Sensuctl and sensu-backend now ask users to retype their passwords when creating a new password in interactive mode.

**FIXES:**

- ([Commercial feature][193]) Sensuctl no longer produces an error when SSL certificates for the Vault provider do not exist on the local system.

> - ([Commercial feature][193]) The [prune API][194] and its [sensuctl interface][195] now need less broad permissions to work.

Author: ```suggestion
- ([Commercial feature][193]) The [prune API][194] and its [sensuctl interface][195] now require less-broad permissions.
```

- Logs now consistently use `event_id` rather than `event_uuid`.
- Sensuctl commands that only contain subcommands now exit with status code 46 when no arguments or incorrect arguments are given.
- The sensuctl dump command now includes a description.
- Sensuctl command descriptions now have consistent capitalization.

> - The `config-file` flag is no longer order-dependent.

Reviewer: ```suggestion
- Use of the `config-file` flag is no longer order-dependent.
```
